source/common/network/address_impl.cc

```diff
@@ -8,30 +8,86 @@
 namespace Network {
 namespace Address {
 
+Address::InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t ss_len) {
+  if (ss_len != 0 && ss_len <= sizeof(sa_family_t)) {
```

Comment:
In general, my preference is to only have error handling if it can reasonably happen in prod. Some of the error handling in this function I don't think can reasonably happen beyond programming errors that should be caught by tests and ASSERT(). In general I think it makes the code more concise, less tests to write for good coverage, etc. I will leave it up to you if you want to convert some of the error handling to just ASSERT checks. Not a huge deal either way.